docs/torch/distrib.rst

```diff
@@ -0,0 +1,353 @@
+    ...         config["feature_dim"],
+    ...         hidden_channels=64,
+    ...         num_layers=2,
+    ...         out_channels=64,
+    ...     ).to(device)
+    ...     model = train.torch.prepare_model(model)
+    ...
+    ...     optimizer = torch.optim.Adam(
+    ...         model.parameters(),
+    ...         lr=0.005 * session.get_world_size(),
+    ...     )
+    ...     train_dataset = SageDataset(
+    ...         batch_size=config["batch_size"],
+    ...         fanout=config["fanout"],
+    ...         graph=config["get_graph"](),
+    ...         feature_dim=config["feature_dim"],
+    ...         label_dim=config["label_dim"],
+    ...         num_nodes=config["num_nodes"],
+    ...         seed_edge_type=0,
+    ...         fanout_edge_types=np.array([0], np.int32),
+    ...         count=10,
+    ...         neg_node_type=[0],
+    ...     )
+    ...
+    ...     pred_train_dataset = SageDataset(
+    ...         batch_size=config["batch_size"],
+    ...         fanout=config["fanout"],
+    ...         graph=config["get_graph"](),
+    ...         feature_dim=config["feature_dim"],
+    ...         label_dim=config["label_dim"],
+    ...         num_nodes=config["num_nodes"],
+    ...         seed_edge_type=0,
+    ...         fanout_edge_types=np.array([0, 1], np.int32),
+    ...         count=10,
+    ...         neg_node_type=[0, 1],
+    ...     )
+    ...
+    ...     test_dataset = SageDataset(
+    ...         batch_size=config["batch_size"],
+    ...         fanout=config["fanout"],
+    ...         graph=config["get_graph"](),
+    ...         feature_dim=config["feature_dim"],
+    ...         label_dim=config["label_dim"],
+    ...         num_nodes=config["num_nodes"],
+    ...         seed_edge_type=1,
+    ...         fanout_edge_types=np.array([0, 1], np.int32),
+    ...         count=10,
+    ...         neg_node_type=[2],
+    ...     )
+    ...     train_dataloader = DataLoader(train_dataset)
+    ...     train_dataloader = train.torch.prepare_data_loader(train_dataloader)
+    ...     pred_train_dataloader = DataLoader(pred_train_dataset)
+    ...     pred_train_dataloader = train.torch.prepare_data_loader(pred_train_dataloader)
+    ...     test_dataloader = DataLoader(test_dataset)
+    ...     test_dataloader = train.torch.prepare_data_loader(test_dataloader)
+    ...
+    ...     test_f1 = 0
+    ...     for epoch in range(config["num_epochs"]):
+    ...         loss = train_model(
+    ...             model,
+    ...             optimizer,
+    ...             train_dataloader,
+    ...         )
+    ...         loss /= train_dataset.count
+    ...         train_f1, test_f1 = eval_model(model, pred_train_dataloader, test_dataloader)
+    ...         get_logger().info(
+    ...             f"Epoch: {epoch:03d}, Loss: {loss:.4f}, Train F1: {train_f1:.4f}, Test F1: {test_f1:.4f}"
+    ...         )
+    ...     session.report({"test_f1": test_f1.item(), "train_f1": test_f1.item()})
```

Comment:
I think for multi-node usage, it is best we use session.report every epoch instead of get_logger().info since we don't want to check logs for each worker we just want the data all in one place with session.report?

Reply:
Makes sense. Less code is also beter.